### PR TITLE
ROE-1105 Ensure single email error

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-1105

### Change description

As with the date fields we want to display only one relevant mail error message at a time

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -295,3 +295,35 @@ const checkCorporatesAddress = (trust: trustType.Trust, addressMaxLength: number
     }
   }
 };
+
+export const validateEmail = (email: string, maxLength: number) => {
+  const isEmailPresent = checkEmailIsPresent(email);
```
Comment (mouhajer-ch) [inline src/validation/custom.validation.ts:300]: You do not need to check it with ifs something like the following will do
```
export const validateEmail = (email: string, maxLength: number) => {
    checkEmailIsPresent(email);
    checkIsWithinLengthLimit(email, maxLength);
    checkCorrectIsFormat(email);
    
    return true;
}
```

File: src/validation/custom.validation.ts [hunk 2]
```diff
@@ -295,3 +295,35 @@ const checkCorporatesAddress = (trust: trustType.Trust, addressMaxLength: number
...
+    const isWithinLengthLimit = checkIsWithinLengthLimit(email, maxLength);
+    if (isWithinLengthLimit) {
+      checkCorrectIsFormat(email);
+    }
+  }
+  return true;
+};
+
+const checkEmailIsPresent = (email: string) => {
+  if (email === undefined || email === "") {
```
Comment (mouhajer-ch) [inline src/validation/custom.validation.ts:311]: Probably email needs to be trimmed before checking it.

File: src/validation/fields/email.validation.ts [hunk 3]
```diff
@@ -1,17 +1,12 @@
...
-    .matches(VALID_EMAIL_FORMAT).withMessage(ErrorMessages.EMAIL_INVALID_FORMAT)
+    .custom((value, { req }) => validateEmail(req.body["email"], 256))
 ];
 
 export const contact_email_validations = [
   body("contact_email").trim()
-    .not().isEmpty().withMessage(ErrorMessages.EMAIL)
-    .isLength({ max: 250 }).withMessage(ErrorMessages.MAX_EMAIL_LENGTH)
-    .matches(VALID_EMAIL_FORMAT).withMessage(ErrorMessages.EMAIL_INVALID_FORMAT)
+    .custom((value, { req }) => validateEmail(req.body["contact_email"], 250))
```
Comment (mouhajer-ch) [inline src/validation/fields/email.validation.ts:11]: Probably I will confirm with analyst why the difference between email length, I can not find the file related to the length of related chips columns

File: src/validation/custom.validation.ts (hunk 1 above)
Comment (mattch1) [reply src/validation/custom.validation.ts:300]: This seems to work so removed the conditions

File: src/validation/custom.validation.ts (hunk 2 above)
Comment (mattch1) [reply src/validation/custom.validation.ts:307]: Trim the email as it is passed in.

File: src/validation/fields/email.validation.ts (hunk 3 above)
Comment (mattch1) [reply src/validation/fields/email.validation.ts:11]: That's a wider question about emails in the web app; this does not relate tot he work for this.

File: src/validation/custom.validation.ts [hunk 4]
```diff
@@ -295,3 +295,31 @@ const checkCorporatesAddress = (trust: trustType.Trust, addressMaxLength: number
     }
   }
 };
+
+export const validateEmail = (email: string, maxLength: number) => {
+  checkEmailIsPresent(email.trim());
+  checkIsWithinLengthLimit(email.trim(), maxLength);
```
Comment (mouhajer-ch) [inline src/validation/custom.validation.ts:301]: I would not trim it for every call, just create single variable and pass it to the three functions, furthermore we do not need to return true from those function, their values is not used

File: src/validation/fields/email.validation.ts (hunk 3 above)
Comment (mouhajer-ch) [reply src/validation/fields/email.validation.ts:11]: Ok, agreed probably we should rise it to the standup

File: src/validation/custom.validation.ts (hunk 4 above)
Comment (mattch1) [reply src/validation/custom.validation.ts:301]: I'm not sure the trim call really matters either way. We cannot remove return true as the validation that calls this requires it. You can verify this by running the unit tests without returning true.

Comment (mattch1) [reply src/validation/custom.validation.ts:301]: Updated the trim to use a variable.

Comment (mouhajer-ch) [reply src/validation/custom.validation.ts]: `return true` should be removed from `checkEmailIsPresent`, `checkIsWithinLengthLimit` and `checkCorrectIsFormat`, it is not used, we need to `return true` for the main email custom validator,  if successfu,l as expected by `express-validator`.

Comment (mattch1) [reply src/validation/custom.validation.ts]: Updated this.

**Review by mouhajer-ch — APPROVED** [review]

**Pull request merged**